cpp/src/Slice/Parser.cpp

```diff
@@ -3318,45 +3318,53 @@ Slice::Operation::outParameters() const
 }
 
 ParameterList
-Slice::Operation::sortedReturnAndOutParameters(const string& returnsName)
+Slice::Operation::returnAndOutParameters(const string& returnsName)
```

Comment:
Split this into 2 separate functions.
`returnAndOutParameters` returns a list containing... the return and out parameters.
`sortedReturnAndOutParameters` calls this and sorts the list into a specific order.

The old `sortedReturnAndOutParameters` did both of these, but there's some languages where we want to preserve the original order.